### PR TITLE
Add new addressing type ADDRESS_REG_REG support. 

### PR DESCRIPTION
**Pull request by chenglulu326**

### CODE REVIEW

File: gcc/config/loongarch/loongarch.cc [hunk 1]
```diff
@@ -1804,6 +1808,35 @@ loongarch_valid_offset_p (rtx x, machine_mode mode)
   return true;
 }
 
+static bool
+loongarch_vaild_index_p (struct loongarch_address_info *info, rtx x,
```
Comment (xen0n) [inline gcc/config/loongarch/loongarch.cc:1812]: valid

File: gcc/config/loongarch/loongarch.cc [hunk 2]
```diff
@@ -3410,12 +3517,28 @@ loongarch_output_move (rtx dest, rtx src)
...
+	  return dbl_p ? "fst.d\t%1,%0" : "fst.s\t%1,%0";
+	}
     }
   if (dest_code == REG && FP_REG_P (REGNO (dest)))
     {
       if (src_code == MEM)
-	return dbl_p ? "fld.d\t%0,%1" : "fld.s\t%0,%1";
+	{
+	  const char *insn = NULL;
+	  insn = loongarch_output_move_index_float(XEXP (src, 0), GET_MODE (src), true);
```
Comment (xen0n) [inline gcc/config/loongarch/loongarch.cc:3536]: 括号前面空格

File: gcc/doc/md.texi [hunk 3]
```diff
@@ -2761,6 +2761,8 @@ A floating-point register (if available).
 A constant call plt address.
 @item j
 A rester that is used as sibing call.
+@item k
+A memory operand whose address is formed by a base register and offset register.
```
Comment (xen0n) [inline gcc/doc/md.texi:2765]: and an (optionally scaled) offset register？

**Review by xen0n — COMMENTED** [review]
手机上随便看了看

**Comment by chenglulu326** [comment]
bot test this please

**Comment by chenglulu326** [comment]
bot test this please

File: gcc/config/loongarch/loongarch.cc (hunk 2 above)
Comment (chenglulu326) [reply gcc/config/loongarch/loongarch.cc]: 额，抱歉，我刚跑了一下代码格式检查！

**Pull request merged**